### PR TITLE
AR testcases for static AR stream: extended and added tests

### PR DESCRIPTION
**Pull request by Sabrina85**

### CODE REVIEW

File: src/odemis/acq/test/stream_test.py [hunk 1]
```diff
@@ -3383,12 +3386,16 @@ def on_im(im):
             self.fail("Failed to find a second point in AR")
 
         e.wait()
+
+        time.sleep(5)  # TODO e.wait() seems to be not to be enough...why?? And why so long?
```
Comment (pieleric) [inline src/odemis/acq/test/stream_test.py]: That's a good question indeed. One possibility is that .image is updated more than twice (ie, `on_im()` is called more than twice). Maybe the first two times, it's updated with the original image, and then later with the new image. You could check relatively easily by adding a log message in `on_im()`.
I've tried on the current version of Odemis (ie, without polarymetry update), and it also happens.
Short tests show that every image update is notified twice.

File: src/odemis/acq/test/stream_test.py [hunk 2]
```diff
@@ -3383,12 +3386,16 @@ def on_im(im):
...
         im2d1 = ars.image.value
         # Check it's a RGB DataArray
         self.assertEqual(im2d1.shape[2], 3)
-
         self.assertFalse(im2d0 is im2d1)
 
+        with self.assertRaises(AssertionError):
+            numpy.testing.assert_array_equal(im2d0, im2d1)
```
Comment (pieleric) [inline src/odemis/acq/test/stream_test.py]: This feels a little odd to read. How about:
`assert not numpy.array_equal(im2d0, im2d1)`

File: src/odemis/acq/test/stream_test.py [hunk 3]
```diff
@@ -3399,12 +3406,16 @@ def on_im(im):
         numpy.testing.assert_equal(ars.background.value, calib[0, 0, 0])
         e.wait()
 
+        time.sleep(3)  # TODO e.wait() seems to be not to be enough...why??
```
Comment (pieleric) [inline src/odemis/acq/test/stream_test.py]: Now, this is getting really weird indeed...

**Comment by Sabrina85** [comment]
.image is still updated multiple times. Leave for now. Put a short time.sleep after getting the .image to make sure we have the last updated version of .image instead of running into trouble requesting the .image while the previous updating of the .image is still in progress. 
When updating the .image twice, the e.wait() is already passed and the new value of .image was requested and compared to the previous one. However, this was still the old image due to the multiple updates of .image.

File: src/odemis/acq/test/stream_test.py [hunk 4]
```diff
@@ -3565,14 +3594,57 @@ def on_im(im):
...
+
+        ###################################################################
+        # test 1 large image + corresponding bg image to test resizing
+        data = [model.DataArray(1500 + numpy.zeros((2000, 2200), dtype=numpy.uint16), metadata[0])]
+        data[0][200:250, 50:70] = 1000  # modify a few px close to AR_POLE
+        logging.info("setting up stream")
+        ars = stream.StaticARStream("test arpol static stream with large image", data)
+
+        # wait a bit for the image to update
+        e = threading.Event()
```
Comment (pieleric) [inline src/odemis/acq/test/stream_test.py]: You don't need to redefine `threading.Event` and `om_im()` in every part of the test case. That said, it seems like this test case could be broken up into three test cases, as each part is independent of the other ones. Could you do of these 2 alternatives?

**Pull request merged**